server/service.go

```diff
@@ -0,0 +1,136 @@
+package server
+
+import (
+    "context"
+    "errors"
+    "sync"
+    "time"
+
+    "github.com/google/uuid"
+    "example.com/student-service/proto" 
+    "google.golang.org/protobuf/types/known/emptypb"
+    "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Student struct {
+    ID        string
+    FirstName string
+    LastName  string
+    Grade     int32
+    CreatedAt time.Time
+}
+
+type Store struct {
+    mu       sync.RWMutex
+    students map[string]Student
+}
+
+type StudentServer struct {
+    proto.UnimplementedStudentServiceServer
+    store *Store
+}
+
+func NewStudentServer() *StudentServer {
+    return &StudentServer{
+        store: &Store{
+            students: make(map[string]Student),
+        },
+    }
+}
+
+func (s *StudentServer) CreateStudent(ctx context.Context, req *proto.CreateStudentRequest) (*proto.CreateStudentResponse, error) {
```

Comment:
StudentServer.CreateStudent -> StudentServer.Create and so on, why double write the entity